substrate/frame/treasury/src/lib.rs

```diff
@@ -292,6 +292,11 @@ pub mod pallet {
 
 	#[pallet::extra_constants]
 	impl<T: Config<I>, I: 'static> Pallet<T, I> {
+		/// The name of the block number provider configured for this pallet.
+		fn block_number_provider_name() -> &'static str {
+			T::BlockNumberProvider::get()
+		}
+
```

Comment:
The point was that this is done for every pallet using `BlockNumberProvider` and that you probably do something like:
```
#[constant]
type BlockNumberProvider...
```

in the `Config` trait.

Reply:
No. The name should be fetched from the BlockNumberProvider type that is already passed to the config.

Reply:
this has been addressed @bkchr